File: test/benchmark/larceny/array1.js
// Larceny Benchmark: array1 (Node.js)
// Array creation, fill, and summation
'use strict';

function main() {
    const __t0 = performance.now();
    const size = 10000;
    const arr = new Int32Array(size);

    for (let i = 0; i < size; i++) {
        arr[i] = i;
    }

    let total = 0;
    for (let iter = 0; iter < 100; iter++) {
        let s = 0;
        for (let i = 0; i < size; i++) {
            s += arr[i];
        }
        total = s;
    }
    const __t1 = performance.now();
    if (total === 49995000) {
        process.stdout.write("array1: PASS\n");
    } else {
        process.stdout.write("array1: FAIL result=" + total + "\n");
    }
    process.stdout.write("__TIMING__:" + (__t1 - __t0) + "\n");
}

main();

File: test/benchmark/larceny/python/array1.py
#!/usr/bin/env python3
# Larceny Benchmark: array1 (Python)
# Array creation, fill, and summation
import time
import array


def main():
    size = 10000
    arr = array.array('i', range(size))

    t0 = time.perf_counter_ns()
    total = 0
    for _ in range(100):
        s = 0
        for i in range(size):
            s += arr[i]
        total = s
    t1 = time.perf_counter_ns()

    if total == 49995000:
        print("array1: PASS")
    else:
        print(f"array1: FAIL result={total}")
    print(f"__TIMING__:{(t1 - t0) / 1e6:.3f}")


main()
